model/builder/action.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 The Serverless Workflow Specification Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"github.com/serverlessworkflow/sdk-go/v2/model"
+)
+
+func NewActionBuilder(name string) *ActionBuilder[any] {
+	return newActionBuilder[any](nil, name)
+}
+
+func newActionBuilder[T any](parent *T, name string) *ActionBuilder[T] {
```

Comment:
Just curious, 'cause I'm kinda of skeptical about generics. Do we really need it since it's always a `WorkflowBuilder`. 🤔

Reply:
It's not really necessary for all cases, but `transition` can have different parents, so generics are needed.

This generics is for the .Parent() method to work